doc/api/addons.markdown

```diff
@@ -85,37 +105,109 @@ to build your module in a JSON-like format. This file gets compiled by
-You can now use the binary addon in a Node.js project `hello.js` by pointing
-`require` to the recently built `hello.node` module:
+When using `npm install` to install a Node.js Addon, npm uses its own bundled
+version of `node-gyp` to perform this same set of actions, generating a
+compiled version of the Addon for the user's platform on demand.
+
+Once built, the binary Addon can be used from within Node.js by pointing
+`require()` to the built `addon.node` module:
 
     // hello.js
     const addon = require('./build/Release/addon');
 
     console.log(addon.hello()); // 'world'
 
-Please see patterns below for further information or
+Please see the examples below for further information or
 <https://github.com/arturadib/node-qt> for an example in production.
 
-
-## Addon patterns
-
-Below are some addon patterns to help you get started. Consult the online
-[v8 reference][] for help with the various v8 calls, and v8's
-[Embedder's Guide][] for an explanation of several concepts used such as
-handles, scopes, function templates, etc.
-
-In order to use these examples, you need to compile them using `node-gyp`.
-Create the following `binding.gyp` file:
+Because the exact path to the compiled Addon binary can vary depending on how
+it is compiled (i.e. sometimes it may be in `./build/Debug/`), Addons can use
+the [bindings][] package to load the compiled module.
+
+Note that while the `bindings` package implementation is more sophisticated
+in how it locates Addon modules, it is essentially using a try-catch pattern
+similar to:
+
+    try {
+      return require('./build/Release/addon.node');
+    } catch (err) {
+      return require('./build/Debug/addon.node');
+    }
+ 
+### Linking to Node.js' own dependencies
+
+Node.js uses a number of statically linked libraries such as v8, libuv and
+OpenSSL. All Addons are required to statically link to v8 and may link to any
+of the other dependencies as well. Typically, this is as simple as including
+the appropriate `#include <...>` statements (e.g. `#include <v8.h>`) and
+`node-gyp` will locate the appropriate headers automatically. However, there
+are a few caveats to be aware of:
+
+* When `node-gyp` runs, it will detect the specific release version of Node.js
+and download either the full source tarball or just the headers. If the full
+source is downloaded, Addons will have complete access to the full set of
+Node.js dependencies. However, if only the Node.js headers are downloaded, then
+only the symbols exported by Node.js will be available.
+
+* `node-gyp` can be run using the `--nodedir` flag pointing at a local Node.js
+source image. Using this option, the Addon will have access to the full set of
+dependencies.
+
+### Loading Addons using require()
+
+The filename extension of the compiled Addon binary is `.node` (as opposed
+to `.dll` or `.so`). The `require()` function is written to look for files
+with the `.node` file extension and initialize those as dynamically-linked
+libraries.
```

Comment:
The previous paragraph says `All Addons are required to statically link to v8 and may link to any of the other dependencies as well.` So is it really loaded as dynamically linked libraries? #justconfirming